### DISCUSSION TITLE
Looks who's back with "Link Aggregation"

### DISCUSSION THREAD

**Discussion by iamnorbiato** [1000001101000/Debian_on_Buffalo#191] [General]
Hi there my friend ! how are you ?

I was a little bit "away" once I had (or I thought that I had) an issue with bay #4 then was running with 3 disks only, a few days ago my no break did not work (battery problems) and I tough that I had loose my device.
Guess what - after a full disassemble, clean up and re-assemble all bays are working again (one disk is gone but replaced) and during the raid5 rebuild (I have 2 backup options now - external disks and cloud for the most important ones).

But here is the discussion:
I have one USB to Ethernet adapter (from my old MacBook) and I plugged in the USB 2.0 and it is working. After an "update" the IP address is also working - Can I configure a "Link Aggregation" or "Bonding" did you tried that already ?

**Comment by 1000001101000** [comment]
A lot of old usb 2.0 NICs are 10/100 which is 10% as fast as the onboard NIC. Even a 10/100/1000 NIC would be less than 50% as fast because of the speed limitation of USB 2.0.

USB 3.0 is a bit different. If we’re talking about the LS441 the USB3 port is connected via a single PCIe 1.0 lane which can theoretically handle more like 2000mb/s. I tried this on mine once with a 2.5gb/s NIC and was able to get around 1.5gb/s but also was taking nearly 100% cpu.

The better question is “what do you hope to achieve with a bonded connection?”. You can certainly set it up but it only provides benefits for certain use cases. A headless device like a linkstation isn’t great for testing network configuration since it’s easy to foul up your config and get locked out of the device.

in summary:
- there is nothing stopping you from doing so
- Anything attached via USB2 will be slow compared to the built-in NIC
- USB3 NIC can be as fast as the built-in and even slightly faster but will add a lot of CPU overhead.
- Testing bonding configurations on headless devices can be difficult 
- The benefits of bonding interfaces depend heavily on the use-case
